dspace-api/src/test/java/org/dspace/statistics/export/ITRetryFailedOpenUrlTracker.java

```diff
@@ -0,0 +1,183 @@
+                failedOpenURLTrackerService.remove(context, tracker);
+            }
+
+            // Clear the list of processedUrls
+            testProcessedUrls.clear();
+
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        } finally {
+            try {
+                context.complete();
+            } catch (SQLException e) {
+                log.error(e);
+            }
+        }
+        super.destroy();
+    }
+
+    /**
+     * Test the mode of the script that allows the user to add a failed url to the database
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testAddNewFailedUrl() throws Exception {
+
+        TestDSpaceRunnableHandler testDSpaceRunnableHandler = new TestDSpaceRunnableHandler();
+        ScriptConfiguration retryOpenUrlTrackerConfig = scriptService.getScriptConfiguration("retry-tracker");
+        DSpaceRunnable retryOpenUrlTracker =
+                scriptService.createDSpaceRunnableForScriptConfiguration(retryOpenUrlTrackerConfig);
+        String urlToAdd = "test-failed-url";
+        String[] args = {"-a", urlToAdd};
+
+        retryOpenUrlTracker.initialize(args, testDSpaceRunnableHandler, eperson);
+        retryOpenUrlTracker.internalRun();
+
+        List<OpenURLTracker> all = failedOpenURLTrackerService.findAll(context);
+
+        assertThat(testProcessedUrls.size(), is(0));
+        assertThat(all.size(), is(1));
+        assertThat(all.get(0).getUrl(), is(urlToAdd));
+    }
+
+    /**
+     * Test to check that all logged failed urls are reprocessed succesfully and removed from the db
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testReprocessAllUrls() throws Exception {
+
+        TestDSpaceRunnableHandler testDSpaceRunnableHandler = new TestDSpaceRunnableHandler();
+        ScriptConfiguration retryOpenUrlTrackerConfig = scriptService.getScriptConfiguration("retry-tracker");
+        DSpaceRunnable retryOpenUrlTracker =
+                scriptService.createDSpaceRunnableForScriptConfiguration(retryOpenUrlTrackerConfig);
+        String[] args = {"-r"};
+
+        OpenURLTracker tracker1 = failedOpenURLTrackerService.create(context);
+        tracker1.setUrl("test-url-1");
+        OpenURLTracker tracker2 = failedOpenURLTrackerService.create(context);
+        tracker2.setUrl("test-url-2");
+        OpenURLTracker tracker3 = failedOpenURLTrackerService.create(context);
+        tracker3.setUrl("test-url-3");
+
+
+        retryOpenUrlTracker.initialize(args, testDSpaceRunnableHandler, eperson);
+        retryOpenUrlTracker.internalRun();
+
+        List<OpenURLTracker> all = failedOpenURLTrackerService.findAll(context);
+
+        assertThat(testProcessedUrls.size(), is(3));
+        assertThat(testProcessedUrls.contains("test-url-1"), is(true));
+        assertThat(testProcessedUrls.contains("test-url-2"), is(true));
+        assertThat(testProcessedUrls.contains("test-url-3"), is(true));
```

Comment:
Just a note.  There's an `assertEquals()` method instead of `assertThat(..., is())`  There's also an `assertTrue()` instead of `assertThat(... is(true))`.  Those alternatives are a bit easier to read.  But, if you'd rather just use `assertThat()` everywhere, that's OK.